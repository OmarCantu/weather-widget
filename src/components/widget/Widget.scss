.widget {
  
}

.icon {

}

.city {

}

.temperature {

}

.wind {

}

.wind-word {

}