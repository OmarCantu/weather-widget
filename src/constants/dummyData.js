export const DEFAULT_WEATHER_OBJECT = {
  main: {
    temp: 'NA'
  },
  name: 'NA',
  weather: [{
    description: 'NA',
    icon: 'NA'
  }],
  wind: {
    speed: 'NA',
    deg: 'NA'
  }
}

export const DUMMY_WEATHER_OBJECT = {
  main: {
    temp: '22'
  },
  name: 'Sydney',
  weather: [{
    description: 'sunny',
    icon: 'NA'
  }],
  wind: {
    speed: '10',
    deg: '0'
  }
}
