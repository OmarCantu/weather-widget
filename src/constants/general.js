export const FILTER_DATA = [
  { category: 'industry', name: 'Industries' }, 
  { category: 'location', name: 'Location' },
  { category: 'company_size', name: 'Company Size' },
  { category: 'use_case', name: 'Use Case' }
];

export const READ_WORDS_PER_MINUTE = 200;