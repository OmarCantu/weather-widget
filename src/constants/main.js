export const APP_ID = 'b6f782215dc85bde2e4fb339d7903f7d'; // Personal id for Open Weather API
export const DEFAULT_GEOLOCATION_URL = 'https://samples.openweathermap.org/data/2.5/weather?q=London,uk&appid=b6907d289e10d714a6e88b30761fae22';
export const DEFAULT_WIDGET_TITLE = 'Title of widget';
export const IMPERIAL= 'imperial';
export const METRIC = 'metric';
export const OFF = 'off';
export const ON = 'on';
export const TEMPERATURE = 'temperature';
export const TITLE = 'title';
export const WIND = 'wind';