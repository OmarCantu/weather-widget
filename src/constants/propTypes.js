import PropTypes from 'prop-types';

export const COMPANY = PropTypes.shape({
  company_id: PropTypes.number.isRequired,
  company_name: PropTypes.string.isRequired,
  image_url: PropTypes.string.isRequired,
  excerpt: PropTypes.string.isRequired,
  industry: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  company_size: PropTypes.string.isRequired,
  use_case: PropTypes.array.isRequired,
  word_count: PropTypes.number.isRequired
});

COMPANY.default = {}

      
