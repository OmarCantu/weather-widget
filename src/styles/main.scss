@import "_colors.scss";
@import "_media.scss";
@import "_resets.scss";
@import "_sizes.scss";
@import "_utils.scss";
