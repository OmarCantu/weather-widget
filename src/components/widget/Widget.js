import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import classNames from 'classnames';

import Weather from '../../classes/Weather';
import { OFF } from '../../constants/main';

import styles from './Widget.scss';

const Widget = props => {
  const { className, data, title, units, windOption } = props;

  useEffect(() => {
    setWeather(new Weather(data, units));
  }, [data]);

  const [weather, setWeather] = useState(new Weather(data, units), []);

  const componentClassName = classNames(styles.widget, className);
  const windClassName = classNames(styles.wind, {
    [styles.hidden]: windOption === OFF
  });

  return (
    <section className={componentClassName}>
      <h2 className={styles.title}>{title}</h2>

      <div className={styles.content}>
        <img 
          alt={weather.description()} 
          className={styles.icon} 
          role='presentation'
          src={weather.iconSrc()} 
          title={weather.description()} 
        />

        <div className={styles.details}>
          <h3 className={styles.city}>{weather.city}</h3>

          <strong className={styles.temperature}>
            {weather.formattedTemp()}
          </strong>

          <div 
            aria-hidden={windOption === OFF}
            className={windClassName}
          >
            <label 
              className={styles['wind-word']}
              htmlFor='wind-details'
            >
              Wind
            </label>

            <div id='wind-details'>
              <span className={styles['wind-direction']}>
                {weather.wind.formattedDirection()}
              </span>

              <span className={styles['wind-speed']}>
                {weather.wind.formattedSpeed()}
              </span>
            </div>
          </div>
        </div>
      </div>       
    </section>
  )
}

Widget.propTypes = {
  className: PropTypes.string, 
  data: PropTypes.object,
  title: PropTypes.string,
  units: PropTypes.string,
  windOption: PropTypes.string
};

Widget.defaultProps = {
  className: undefined,
  data: undefined,
  title: undefined,
  units: undefined,
  windOption: undefined
};

export default Widget;