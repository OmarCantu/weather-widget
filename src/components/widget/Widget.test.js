import React from 'react';
import { shallow } from 'enzyme';

import { DUMMY_WEATHER_OBJECT } from '../../constants/dummyData';
import { METRIC, ON } from '../../constants/main';

import Widget from './Widget';

const wrapper = shallow(
  <Widget 
    className='custom'
    data={DUMMY_WEATHER_OBJECT}
    title='Widget'
    units={METRIC}
    windOption={ON}
  />
);

test('Widget has a "custom" class', () => {
  expect(wrapper.find('.custom')).toHaveLength(1);
});

test('Widget has one h2 tag with content "Widget"', () => {
  expect(wrapper.find('h2').text()).toEqual('Widget');
});

test('Widget has one img tag with alt "sunny"', () => {
  expect(wrapper.find('img[alt="sunny"]')).toHaveLength(1);
});

test('Widget has one h3 tag with content "Sydney"', () => {
  expect(wrapper.find('h3').text()).toEqual('Sydney');
});

test('Widget has one strong tag with content "22°"', () => {
  expect(wrapper.find('strong').text()).toEqual('22°');
});

test('Component matches Snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});