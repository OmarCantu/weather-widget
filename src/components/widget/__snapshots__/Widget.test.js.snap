// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches Snapshot 1`] = `ShallowWrapper {}`;
